client/components/editor-standalone/authoring-storage-in-memory.ts

```diff
@@ -1,17 +1,31 @@
 import ng from 'superdesk-core/scripts/core/services/ng';
 import {IAuthoringAutoSave, IAuthoringStorage} from 'superdesk-api';
 import {getProfile} from './profile';
+import {autosave} from '../../api/autosave';
 
-export function getAuthoringStorageInMemory<T>(
+export function getAuthoringStorageInMemory<T extends {_id: string}>(
     profile: 'event' | 'planning',
     item: T,
     onSave: (current: T, original: T) => Promise<T>,
 ): IAuthoringStorage<T> {
+    /**
+     * Timeout for 500 seconds to let newly added embedded events finish autosaving.
+     */
     class NoAutoSave implements IAuthoringAutoSave<T> {
         get(id: string) {
-            // return a different reference so authoring-react sees it as having unsaved changes.
-            // otherwise, it will consider all saved and won't attempt to save - which won't trigger validation.
-            return Promise.resolve({...item});
+            return new Promise<T>((resolve) => {
+                setTimeout(() => {
+                    autosave
+                        .getById(profile === 'event' ? 'event' : 'planning', id)
```

Comment:
> profile === 'event' ? 'event' : 'planning'

It's a trap to do it like that. If more items are added to the union type the code will fail silently. And we want it to fail loud and clear :D Use if/else if/assert never